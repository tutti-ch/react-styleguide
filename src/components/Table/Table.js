import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import classes from './Table.scss'

export default class Table extends Component {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.oneOfType([PropTypes.string, PropTypes.object, PropTypes.array])
  }

  static Caption = props => <caption {...props}>{props.children}</caption>
  static Body = ({ children }) => <tbody>{children}</tbody>
  static Head = ({ children }) => <thead>{children}</thead>
  static Foot = ({ children }) => <tfoot>{children}</tfoot>
  static Row = ({ children }) => <tr>{children}</tr>
  static Cell = props => <td {...props}>{props.children}</td>
  static HCell = props => <th {...props}>{props.children}</th>

  render() {
    const { className, children } = this.props

    return (
      <table className={classNames(className, classes.table)}>
        {children}
      </table>
    )
  }
}