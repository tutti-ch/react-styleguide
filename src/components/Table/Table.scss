@import "./table";

// Required for Readme.md
:global .table-cell-styled {
  font-weight: bold;
  color: $green-50;
}