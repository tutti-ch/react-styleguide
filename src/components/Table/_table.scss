@import "main";

table {
  margin: 0;
  width: 100%;
  table-layout: fixed;
  border-collapse: collapse;

  @include tablet-and-bigger {
    background: $greyscale-100; // Needed to fill empty cells in two column layout
  }
}

caption {
  padding-bottom: $size-m;
  text-align: left;
  font-size: $size-ml;
}

th,
td {
  border: 1px solid $greyscale-90;
  padding: $size-l;
}

th {
  font-weight: bold;
  text-align: center;
}

@include phablet-and-smaller {
  table,
  thead,
  tbody,
  tfoot,
  tr,
  caption {
    display: block;
  }

  td,
  th {
    box-sizing: border-box;
    display: flex;
    align-items: center;
    width: 100%;
    border-top: none;
  }

  tr:first-child {
    td:first-child,
    th:first-child {
      border-top: 1px solid $greyscale-90;
    }
  }

  tbody + tfoot tr:first-child td:first-child,
  thead + tbody tr:first-child td:first-child {
    border-top: none;
  }
}
