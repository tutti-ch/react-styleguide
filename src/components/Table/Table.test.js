/* global describe, expect, test, jest */
import React from 'react'
import Table from './index'
import renderer from 'react-test-renderer'

describe('(Component) Table', () => {
  test('should render correctly', () => {
    const tree = renderer.create(
      <Table>
        <Table.Caption>This is the table caption</Table.Caption>
        <Table.Head>
          <Table.Row>
            <Table.HCell colSpan={2}>Wide column</Table.HCell>
            <Table.HCell>Narrow column</Table.HCell>
          </Table.Row>
        </Table.Head>
        <Table.Body>
          <Table.Row>
            <Table.Cell>R1C1</Table.Cell>
            <Table.Cell>R1C2</Table.Cell>
            <Table.Cell>R1C3</Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>R2C1</Table.Cell>
            <Table.Cell>R2C2</Table.Cell>
            <Table.Cell>R2C3</Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>R3C1</Table.Cell>
            <Table.Cell>R3C2</Table.Cell>
            <Table.Cell>R3C3</Table.Cell>
          </Table.Row>
        </Table.Body>
        <Table.Foot>
          <Table.Row>
            <Table.Cell style={{ color: 'red' }}>Narrow column</Table.Cell>
            <Table.Cell colSpan={2} className='table-cell-styled'>Extra styles</Table.Cell>
          </Table.Row>
        </Table.Foot>
      </Table>
    )

    expect(tree).toMatchSnapshot()
  })
})
