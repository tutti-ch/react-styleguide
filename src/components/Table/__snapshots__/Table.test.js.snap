// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Table should render correctly 1`] = `
<Table>
  <table
    className=""
  >
    <Component>
      <caption>
        This is the table caption
      </caption>
    </Component>
    <Component>
      <thead>
        <Component>
          <tr>
            <Component
              colSpan={2}
            >
              <th
                colSpan={2}
              >
                Wide column
              </th>
            </Component>
            <Component>
              <th>
                Narrow column
              </th>
            </Component>
          </tr>
        </Component>
      </thead>
    </Component>
    <Component>
      <tbody>
        <Component>
          <tr>
            <Component>
              <td>
                R1C1
              </td>
            </Component>
            <Component>
              <td>
                R1C2
              </td>
            </Component>
            <Component>
              <td>
                R1C3
              </td>
            </Component>
          </tr>
        </Component>
        <Component>
          <tr>
            <Component>
              <td>
                R2C1
              </td>
            </Component>
            <Component>
              <td>
                R2C2
              </td>
            </Component>
            <Component>
              <td>
                R2C3
              </td>
            </Component>
          </tr>
        </Component>
        <Component>
          <tr>
            <Component>
              <td>
                R3C1
              </td>
            </Component>
            <Component>
              <td>
                R3C2
              </td>
            </Component>
            <Component>
              <td>
                R3C3
              </td>
            </Component>
          </tr>
        </Component>
      </tbody>
    </Component>
    <Component>
      <tfoot>
        <Component>
          <tr>
            <Component
              style={
                Object {
                  "color": "red",
                }
              }
            >
              <td
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                Narrow column
              </td>
            </Component>
            <Component
              className="table-cell-styled"
              colSpan={2}
            >
              <td
                className="table-cell-styled"
                colSpan={2}
              >
                Extra styles
              </td>
            </Component>
          </tr>
        </Component>
      </tfoot>
    </Component>
  </table>
</Table>
`;
